File: scm-csharp/scheme/primitives/PrimitiveThreadYieldB.cs
using System.Threading;

namespace scheme;

public class PrimitiveThreadYieldB : Primitive
{
    public override string Name() => "thread-yield!";

    public override string Info() =>
        "Syntax: (thread-yield!)\n" +
        "Library: (srfi 18)\n" +
        "Description: Causes the current thread to yield the processor to other threads.\n" +
        "Example:\n" +
        "  (thread-yield!)";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 0, 0);
        Thread.Yield();
        return Value.NIL;
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveThreadYieldB.java
package scheme.primitives;
import scheme.*;

public class PrimitiveThreadYieldB extends Primitive {
    @Override
    public String name() { return "thread-yield!"; }

    @Override
    public String info() {
        return "Syntax: (thread-yield!)\n" +
               "Library: (srfi 18)\n" +
               "Description: Causes the current thread to yield the processor.\n" +
               "Example:\n" +
               "  (thread-yield!)";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 0, 0);
        Thread.yield();
        return Value.NIL;
    }
}
